packages/ecs/src/topic.ts

```diff
@@ -0,0 +1,52 @@
+export type Topic<T extends string = string, D = any> = {
+  /**
+   * A property for the name of the topic
+   */
+  name: string
+
+  /**
+   * Provides iterator syntax to the consumers of a topic. This will loop
+   * through all of the events with the type specified in the second type 
+   * parameter. 
+   */
+  [Symbol.iterator](): IterableIterator<D>;
+
+  /**
+   * Pushes an event to the topic so that it can be consumed by downstream 
+   * systems.
+   */
+  pushEvent(event: D): void
```

Comment:
Could potentially make this `pushEvents(...events: D[])` since Array.push supports the rest operator.